clients/vscode/src/inline-edit/util.ts

```diff
@@ -0,0 +1,75 @@
+export interface InlineEditParseResult {
+  /**
+   * mentions, start with '@'
+   */
+  mentions?: string[];
+  /**
+   * last mention in the end of user commnad.
+   * for `explain @`, mentionQuery is `''`,  we can trigger file pick
+   * for `explain @file`, mentionQuery is `file`,  we know user is editing the mention
+   * for `explain @file to me`, mentionQuery is `undefined`
+   */
+  mentionQuery?: string;
+}
+
+export interface InlineEditCommand extends InlineEditParseResult {
+  command: string;
+}
```

Comment:
If we save a map of label to file uri in UserCommandQuickpick scope, we will not parse the command again to get the context list.
```suggestion
export interface InlineEditCommand {
  command: string;
  context: {
    referer: string,
    uri: Uri,
  }[];
}
```

And I think `parseUserCommand` could only handle the case of the trailing query, so I could return single string instead of InlineEditParseResult.

Reply:
A map of label to file can't make this `parseUserCommand` just return string, because user may delete file mentions in  command.

Reply:
- added a map of label to file, but still need to do the `parseUserCommand`